### PR TITLE
Genesistls plugin

### PR DESCRIPTION
**Pull request by NaddH1**

### CODE REVIEW

**Comment by nyagami** [comment]
Please remove the changes in `package-lock.json` file

File: plugins/english/genesistl.ts [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+            link += filters.type.value.map((i) => `&lang[]=${i}`).join("");
+
+        link += "&status=" + filters.status.value;
+
+        link += "&order=" + filters.order.value;
+
+        const headers = new Headers();
+        const body = await fetchApi(link, { headers }).then((result) =>
+            result.text()
+        );
```
Comment (nyagami) [inline plugins/english/genesistl.ts:54]:
```suggestion
        const body = await fetchApi(link).then((result) =>
            result.text()
        );
```

File: plugins/english/genesistl.ts [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+
+        const loadedCheerio = parseHTML(body);
+        return this.parseNovels(loadedCheerio);
+    }
+
+    async parseNovelAndChapters(novelUrl: string): Promise<Plugin.SourceNovel> {
+        const url = novelUrl;
+        const headers = new Headers();
+        const result = await fetchApi(url, { headers });
```
Comment (nyagami) [inline plugins/english/genesistl.ts:63]:
```suggestion
        const result = await fetchApi(url);
```

File: plugins/english/genesistl.ts [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+            .each(function () {
+                const chapterName =
+                    loadedCheerio(this).find(".epl-num").text() +
+                    " - " +
+                    loadedCheerio(this).find(".epl-title").text();
+
+                const releaseDate = loadedCheerio(this)
+                    .find(".epl-date")
+                    .text()
+                    .trim();
```
Comment (nyagami) [inline plugins/english/genesistl.ts:123]: please parse this into ISO string format

File: plugins/english/genesistl.ts [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+                });
+            });
+
+        novel.chapters = chapter.reverse();
+
+        return novel;
+    }
+    async parseChapter(chapterUrl: string): Promise<string> {
+        const headers = new Headers();
+        const result = await fetchApi(chapterUrl, { headers });
```
Comment (nyagami) [inline plugins/english/genesistl.ts:144]:
```suggestion
        const result = await fetchApi(chapterUrl);
```

File: plugins/english/genesistl.ts [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+        return chapterText;
+    }
+
+    async searchNovels(
+        searchTerm: string,
+        pageNo: number
+    ): Promise<Plugin.NovelItem[]> {
+        const url = `${this.site}page/${pageNo}/?s=${searchTerm}`;
+        const headers = new Headers();
+        const result = await fetchApi(url, { headers });
```
Comment (nyagami) [inline plugins/english/genesistl.ts:160]:
```suggestion
        const result = await fetchApi(url);
```

File: plugins/english/genesistl.ts [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+    }
+    async popularNovels(
+        pageNo: number,
+        { filters }: Plugin.PopularNovelsOptions<typeof this.filters>
+    ): Promise<Plugin.NovelItem[]> {
+        let link = `${this.site}series/?page=${pageNo}`;
+        if (filters.genres.value.length)
+            link += filters.genres.value.map((i) => `&genre[]=${i}`).join("");
+        if (filters.type.value.length)
+            link += filters.type.value.map((i) => `&lang[]=${i}`).join("");
```
Comment (K1ngfish3r) [inline plugins/english/genesistl.ts:36]:
```suggestion
            link += filters.type.value.map((i) => `&type[]=${i}`).join("");
```

File: plugins/english/genesistl.ts [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+                { label: "Wholesome", value: "wholesome" },
+                { label: "Wuxia", value: "wuxia" },
+                { label: "Xuanhuan", value: "xuanhuan" },
+                { label: "Yandere", value: "yandere" },
+                { label: "Yuri", value: "yuri" },
+            ],
+            type: FilterTypes.CheckboxGroup,
+            value: [],
+        },
+    } satisfies Filters;
```
Comment (K1ngfish3r) [inline plugins/english/genesistl.ts:267]:
```suggestion
    filters = {
        order: {
            label: "Sort By",
            options: [
                { label: "Default", value: "" },
                { label: "A-Z", value: "title" },
                { label: "Z-A", value: "titlereverse" },
                { label: "Latest Update", value: "update" },
                { label: "Latest Added", value: "latest" },
                { label: "Popular", value: "popular" },
            ],
            type: FilterTypes.Picker,
            value: "",
        },
        status: {
            label: "Status",
            options: [
                { label: "All", value: "" },
                { label: "Ongoing", value: "ongoing" },
                { label: "Hiatus", value: "hiatus" },
                { label: "Completed", value: "completed" },
            ],
            type: FilterTypes.Picker,
            value: "",
        },
        type: {
            label: "Type",
            options: [
                { label: "korean novel", value: "korean-novel" },
                { label: "Web Novel", value: "web-novel" },
            ],
            type: FilterTypes.CheckboxGroup,
            value: [],
        },
        genres: {
            label: "Genres",
            options: [
                { label: "Academy", value: "academy" },
                { label: "Action", value: "action" },
                { label: "Adult", value: "adult" },
                { label: "Adventure", value: "adventure" },
                { label: "Another World", value: "another-world" },
                { label: "Comdey", value: "comdey" },
                { label: "Comedy", value: "comedy" },
                { label: "Dark Fantasy", value: "dark-fantasy" },
                { label: "Drama", value: "drama" },
                { label: "Fantasy", value: "fantasy" },
                { label: "Fantasy Fusion", value: "fantasy-fusion" },
                { label: "Harem", value: "harem" },
                { label: "Historical", value: "historical" },
                { label: "Horror", value: "horror" },
                { label: "Hunter", value: "hunter" },
                { label: "Light Novel", value: "light-novel" },
                { label: "Martial Arts", value: "martial-arts" },
                { label: "Mature", value: "mature" },
                { label: "Misunderstanding", value: "misunderstanding" },
                { label: "Modern", value: "modern" },
                { label: "Munchkin", value: "munchkin" },
                { label: "Murim", value: "murim" },
                { label: "mystery", value: "mystery" },
                { label: "No Harem", value: "no-harem" },
                { label: "NO NTR", value: "no-ntr" },
                { label: "obsession", value: "obsession" },
                { label: "Possession", value: "possession" },
                { label: "Psychological", value: "psychological" },
                { label: "Regression", value: "regression" },
                { label: "Regret", value: "regret" },
                { label: "reincarnation", value: "reincarnation" },
                { label: "Romance", value: "romance" },
                { label: "School Life", value: "school-life" },
                { label: "Seinen", value: "seinen" },
                { label: "Slice of life", value: "slice-of-life" },
                { label: "Supernatural", value: "supernatural" },
                { label: "Tragedy", value: "tragedy" },
                { label: "Transmigrated to Game", value: "transmigrated-to-game" },
                { label: "Transmigration", value: "transmigration" },
            ],
            type: FilterTypes.CheckboxGroup,
            value: [],
        },
    } satisfies Filters;
```

**Review by nyagami — APPROVED** [review]

**Pull request merged**